.github/workflows/build-test-distribute.yaml

```diff
@@ -24,20 +28,21 @@ jobs:
         with:
           path: |
             ${{ env.CI_TOOLS_DIR }}
-          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          key: devtools-${{ env.RUN_FULL_MATRIX && 'full-matrix' || 'normal' }}-${{ runner.os }}-${{ hashFiles('mk/dependencies/deps.lock') }}
```

Comment:
Actually dev-tools is the same whether we do full matrix or not. It's only the code cache (from setup-go)

Reply:
Got it, just updated code to implement it.